core/modules/image/image.tokens.inc

```diff
@@ -0,0 +1,324 @@
+                      break;
+
+                    case 'mimetype':
+                      $output[] = $images_style_data[$styled_uri]['mime_type'];
+                      break;
+
+                    case 'extension':
+                      $output[] = $images_style_data[$styled_uri]['extension'];
+                      break;
+
+                    case 'filesize':
+                      $output[] = $images_style_data[$styled_uri]['file_size'];
+                      break;
+                  }
+                  break;
+
+                case 'path':
+                  if (isset($explode[2]) && $explode[2] === 'relative') {
+                    $url  = parse_url(image_style_url($type, $field['uri']));
+                    $output[] = $url['path'];
+                  }
+                  else {
+                    $output[] = image_style_url($type, $field['uri']);
+                  }
+                  break;
+
+                default:
+                  if (isset($field[$attribute])) {
+                    $output[] = $field[$attribute];
+                  }
+              }
+            }
+            else {
+              $output[] = image_style_url($type, $field['uri']);
+            }
+          }
+
+          // Compile each of the elements into a comma separated list, per
+          // standard Token practice.
+          $replacements[$original] = implode(', ', $output);
+        }
+      }
+    }
+  }
+
+  return $replacements;
+}
+
+/**
+ * Helper function for getting all available image attributes.
+ *
+ * @return array
+ *   All of the available image attributes.
+ */
+function _image_style_token_image_attributes() {
+  return array(
+    'style_name' => NULL,
+    'uri' => '',
+    'path' => NULL,
+    'path:relative' => NULL,
+    'width' => NULL,
+    'height' => NULL,
+    'extension' => NULL,
+    'filesize' => NULL,
+    'alt' => '',
+    'title' => NULL,
+    'mimetype' => NULL,
+    'first' => '',
+    'second' => '',
+    'third' => '',
+    'last' => '',
+  );
+}
+
+/**
+ * Helper function to generate image style if does not exist.
+ *
+ * @param string $styled_uri
+ *   URI for image style.
+ * @param string $type
+ *   Machine name for image style.
+ * @param string $field_uri
+ *   URI of field to be styled.
+ */
+function _image_generate_image($styled_uri, $type, $field_uri) {
+  // Only generate the image if it doesn't already exist.
+  if (!file_exists($styled_uri)) {
+    $style = image_style_load($type);
+    // Create a derivative of the image for this style.
+    image_style_create_derivative($style, $field_uri, $styled_uri);
+  }
+}
+
+/**
+ * Helper function to find all image fields.
+ *
+ * @return array
+ *   An array containing all image fields. Keys of the array is an identifier
+ *   made up of field_type:field_name:entity_type:bundle. Values are the
+ *   human-readable name of the field.
+ */
+function _image_get_image_fields() {
```

Comment:
Same here, if this is only used for tokens it should be namespaced to `_image_token_get_image_fields()`

Reply:
renamed.